utils/utils.tsx

```diff
@@ -776,25 +780,25 @@ function updateCodeTheme(userTheme) {
     }
 }
 
-export function placeCaretAtEnd(el) {
+export function placeCaretAtEnd(el: HTMLInputElement | HTMLTextAreaElement) {
     el.focus();
     el.selectionStart = el.value.length;
     el.selectionEnd = el.value.length;
 }
 
-export function getCaretPosition(el) {
+export function getCaretPosition(el: HTMLInputElement | HTMLTextAreaElement) {
     if (el.selectionStart) {
         return el.selectionStart;
-    } else if (document.selection) {
+    } else if ((document as any).selection) {
```

Comment:
You can actually remove this whole block. Every browser since IE9 supports `selectionStart`. Really, this function isn't needed if you'd like to remove it, but I'm 0/5 on doing that right now

Reply:
removed